### PR TITLE
Data objects, assembler, and controller for new Keep API

### PR DESCRIPTION
**Pull request by ryanpbrewster**
@leogrim @andrewconner @camhashemi 

feedback welcome

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    libById.flatMapValues { lib =>
+      val spaceHandle = lib.organizationId.fold[Option[Handle]](userById.get(lib.ownerId).map(_.username))(orgId => orgById.get(orgId).map(_.handle))
+      spaceHandle.map(handle => pathCommander.libPageByHandleAndSlug(handle, lib.slug)).map { libPath =>
+        BasicLibrary(
+          id = Library.publicId(lib.id.get),
+          name = lib.name,
+          path = libPath.relativeWithLeadingSlash,
+          visibility = lib.visibility,
+          color = lib.color
+        )
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala:39]: assuming these fields change much less frequently than this object is consumed, why not cache the object like we do `BasicUser`?

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/assemblers/KeepInfoAssembler.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    assembledInfoFut.map(_.flatMap {
+      case LeftSide(errMsg) =>
+        slackLog.warn(errMsg)
+        None
+      case RightSide((keepInfo, viewerInfo)) =>
+        Some((keepInfo, viewerInfo))
+    })
+  }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/assemblers/KeepInfoAssembler.scala:138]: this is fun, nice going.

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewKeepInfo.scala [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  path: Path,
+  url: String,
+  title: Option[String],
+  imagePath: Option[ImagePath],
+  author: BasicAuthor,
+  keptAt: DateTime,
+  source: Option[SourceAttribution],
+  users: Seq[BasicUser],
+  libraries: Seq[BasicLibrary],
+  recentActivity: Seq[ActivityEvent])
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewKeepInfo.scala]: should be `ActivityLog`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala:39]: I was thinking about that last night, shouldn't be too hard to invalidate (lib changes, username/org handle changes). Definitely a good plan going forward.

Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala:39]: Perhaps use a shorter cache just to be safe as the invalidation is tricky, e.g. 24 hours?

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/assemblers/KeepInfoAssembler.scala [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+object KeepInfoAssemblerConfig {
+  final case class KeepInfoAssemblyOptions(
+    idealImageSize: ImageSize,
+    sanitizeUrls: Boolean)
+
+  val default = KeepInfoAssemblyOptions(
+    idealImageSize = ProcessedImageSize.Large.idealSize,
+    sanitizeUrls = true
+  )
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/assemblers/KeepInfoAssembler.scala:36]: :trophy:

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewKeepInfo.scala [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import com.keepit.social.{ BasicAuthor, BasicUser }
+import org.joda.time.DateTime
+import play.api.libs.json._
+
+case class NewKeepInfo(
+  id: PublicId[Keep],
+  path: Path,
+  url: String,
+  title: Option[String],
+  imagePath: Option[ImagePath],
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewKeepInfo.scala]: We might want the image size here, I wonder we should plan for it and have an `image` field that nests `path`

File: kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContent.scala [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import com.keepit.rover.model.ArticleVersion
+import org.joda.time.DateTime
+
+case class NewPageContent(
+  summary: NewPageSummary,
+  history: Seq[NewPageSummary])
+
+case class NewPageSummary(
+  author: Option[PageAuthor],
+  domainName: Option[String],
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/shoebox/data/keep/NewPageContent.scala]: I think `site` or `siteName` is better than `domainName` here. If you google "domain name" it usually refers to the domain itself.

**Comment by leogrim** [comment]
:+1: 
A bold move would be to rename existing classes and prefix them with `Deprecated`, then you can claim their current names.

**Comment by ryanpbrewster** [comment]
Bold indeed.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/gen/BasicLibraryGen.scala:39]: I guess `path` invalidation is pretty tough. might not be worth it if it'll lead to broken links.

**Pull request merged**